### PR TITLE
nrf_802154_sl: Add initial nRF53 multiprotocol support

### PR DESCRIPTION
**Pull request by pkwiek**
This commit adds initial support for multiprotocol for nRF53.

Signed-off-by: Pawel Kwiek <pawel.kwiek@nordicsemi.no>

### CODE REVIEW

File: nrf_802154_sl/CHANGELOG.rst [hunk 1]
```diff
@@ -12,11 +12,10 @@ All notable changes to this project are documented in this file.
 nRF 802.15.4 Service Layer 0.3.0
```
Comment (hubertmis) [inline nrf_802154_sl/CHANGELOG.rst:12]: It seems we should mark it here as version 0.4.0 with its changelog entry.

File: nrf_802154_sl/CHANGELOG.rst [hunk 2]
```diff
@@ -12,11 +12,10 @@ All notable changes to this project are documented in this file.
...
 ********************************
 
-* PA/LNA implementation has been moved to MPSL.
-  Obsolete implementation and API have been removed.
+PA/LNA implementation has been moved to MPSL. Obsolete implementation and API have been removed.
 
 Removed
-=======
+=====
```
Comment (hubertmis) [inline nrf_802154_sl/CHANGELOG.rst:18]: It looks like a bug (it's too short)

**Review by hubertmis — CHANGES_REQUESTED** [review]
Looks good! Thanks for updating licenses.

I think we could improve changelog for this patch.

File: nrf_802154_sl/CHANGELOG.rst (hunk 2 above)
Comment (pkwiek) [reply nrf_802154_sl/CHANGELOG.rst]: Fixed

File: nrf_802154_sl/CHANGELOG.rst (hunk 1 above)
Comment (pkwiek) [reply nrf_802154_sl/CHANGELOG.rst:23]: Please check if the new entry is appropriate.

**Review by jciupis — APPROVED** [review]

**Review by hubertmis — APPROVED** [review]

**Review by FrancescoSer — CHANGES_REQUESTED** [review]
@pkwiek @jciupis  Could you add some additional detail on how to utilize this service layer? There are 35 files changed in this PR (and about the same number in the previous one that added this SL to nrfxlib) and the doc files are way too synthetic in comparison.

Some info that would be useful:
- Why would any user actually use this? Example use cases of the 4 features that were implemented here
- Examples of how to actually implement them.
- Integration notes?
- Anything else that the user should know to easily use this

**Comment by hubertmis** [comment]
@FrancescoSer, actually the only user of this library is 802.15.4 radio driver. No one else should use it. Because of that I don't think we need documentation here. Maybe just a note like "This library is used only by 802.15.4 radio driver. Don't use it"

**Review by FrancescoSer — APPROVED** [review]
Added distinction between NCS users and other users of the RD

**Comment by hubertmis** [comment]
> Added distinction between NCS users and other users of the RD

Thanks! The description is much clearer now

**Comment by e-rk** [comment]
https://github.com/nrfconnect/sdk-nrf/pull/3719 is green. Please merge.

**Pull request merged**